include/zephyr/net_buf.h

```diff
@@ -1832,6 +1834,96 @@ static inline void net_buf_add_be64(struct net_buf *buf, uint64_t val)
 	net_buf_simple_add_be64(&buf->b, val);
 }
 
+/**
+ * @brief Add a character string to the end of the buffer, with length limit
+ *
+ * Adds a @c \\0 terminated string to buffer if it fits, up to a maximum length.
+ * The terminating @c \\0 itself is not added.
+ *
+ * See @ref net_buf_simple_to_str to access the data as a regular C string.
+ *
+ * @param buf Buffer to update.
+ * @param str String to be added.
+ * @param max_len Maximum length of @c str that will be added to the buffer,
+ *        truncating it if longer.
+ *
+ * @return 0 if the string could fit.
+ * @return -ENOMEM if the buffer was too short for the string.
+ */
+int net_buf_simple_add_str_len(struct net_buf_simple *buf, const char *str, size_t max_len);
+
+/**
+ * @brief Add a character string to the end of the buffer
+ *
+ * Adds a @c \\0 terminated string to buffer if it fits.
+ * The terminating @c \\0 itself is not added.
+ *
+ * See @ref net_buf_simple_to_str to access the data as a regular C string.
+ *
+ * @param buf Buffer to update.
+ * @param str String to be added.
+ *
+ * @return 0 if the string could fit.
+ * @return -ENOMEM if the buffer was too short for the string.
+ */
+static int net_buf_simple_add_str(struct net_buf_simple *buf, const char *str)
```

Comment:
`static` header functions should be declared `inline`

```suggestion
static inline int net_buf_simple_add_str(struct net_buf_simple *buf, const char *str)
```

Reply:
I was wondering why there were warnings! That makes sense, thank you.